feat: Add support for Dart 3.1, remove support for Dart 2.18 (#969)
parse-community/Parse-SDK-Flutter: packages/flutter/README.md

> The Parse Flutter SDK is continuously tested with the most recent release of the Flutter framework to ensure compatibility. To give developers time to upgrade their app to the newest Flutter framework, previous Flutter framework releases are supported for at least 1 year after their [release date](https://docs.flutter.dev/release/archive?tab=linux). The Parse Flutter SDK depends on the Parse Dart SDK which may require a higher Dart framework version than the Flutter framework version, in which case the Flutter framework version cannot be supported even though its release date may have been less than a year ago.

Reviewer: ```suggestion
The Parse Flutter SDK is continuously tested with the most recent release of the Flutter framework to ensure compatibility. To give developers time to upgrade their app to a newer Flutter framework, previous Flutter framework releases are supported for at least 1 year after the [release date](https://docs.flutter.dev/release/archive?tab=linux) of the next higher significant version. The Parse Flutter SDK depends on the Parse Dart SDK which may require a higher Dart framework version than the Flutter framework version, in which case a Flutter framework version cannot be supported.
```

We want to make contributing to this project as easy and transparent as possible. Please refer to the [Contribution Guidelines](https://github.com/parse-community/Parse-SDK-Flutter/blob/master/CONTRIBUTING.md).

> [guide]: https://docs.parseplatform.org/flutter/guide/
> 
> [open-collective-link]: https://opencollective.com/parse-server

Reviewer: ```suggestion
[guide]: https://docs.parseplatform.org/flutter/guide/
[open-collective-link]: https://opencollective.com/parse-server
```
